File: Practica1/Backend-Nodejs/src/controllers/notification.controller.js
const notificationService = require("../services/notification.service");

async function getNotifications(req, res, next) {
  try {
    const { id_usuario } = req.params;

    const notifications = await notificationService.listNotifications(id_usuario);

    res.json({
      success: true,
      unread_count: notifications.filter(n => !n.leida).length,
      total: notifications.length,
      data: notifications
    }); 

  } catch (error) {
    next(error);
  }
}

module.exports = {
  getNotifications
};

File: Practica1/Backend-python/app/controllers/notification_controller.py
from flask import Blueprint, jsonify
from app.services.notification_service import list_notifications

notification_bp = Blueprint("notification_bp", __name__)

@notification_bp.route("/<int:id_usuario>", methods=["GET"])
def get_notifications(id_usuario):
    try:
        notifications = list_notifications(id_usuario)

        return jsonify({
            "success": True,
            "unread_count": len([n for n in notifications if not n["leida"]]),
            "total": len(notifications),
            "data": notifications
        })

    except Exception as e:
        return jsonify({
            "success": False,
            "message": str(e)
        }), 500